posthog/performance/schema.py

```diff
@@ -0,0 +1,185 @@
+
+SELECT * FROM performance_events
+WHERE team_id = 1
+AND session_id = 'my-session-uuid'
+ORDER BY timestamp DESC
+
+## get all performance events for a given team's pageview
+
+allows us to show performance events in a waterfall chart for a given pageview
+
+SELECT * FROM performance_events
+WHERE team_id = 1
+AND pageview_id = 'my-page-view-uuid' -- sent by SDK
+ORDER BY timestamp DESC
+
+## all other queries are expected to be based on aggregating materialized views built from this fact table
+"""
+
+BASE_PERFORMANCE_EVENT_COLUMNS = """
+uuid UUID,
+session_id UUID,
+window_id UUID,
+pageview_id UUID,
+time_origin Int64,
+origin_timestamp DateTime64(3, 'UTC'),
+entry_type LowCardinality(String),
+name String,
+team_id Int64,
+current_url String,
+""".strip()
+
+"""https://developer.mozilla.org/en-US/docs/Web/API/PerformanceResourceTiming"""
+RESOURCE_EVENT_COLUMNS = """
+start_time Int64,
+redirect_start Int64,
+redirect_end Int64,
+worker_start Int64,
+fetch_start Int64,
+domain_lookup_start Int64,
+domain_lookup_end Int64,
+connect_start Int64,
+secure_connection_start Int64,
+connect_end Int64,
+request_start Int64,
+response_start Int64,
+response_end Int64,
+decoded_body_size Int64,
+encoded_body_size Int64,
+initiator_type LowCardinality(String),
+next_hop_protocol LowCardinality(String),
+render_blocking_status LowCardinality(String),
+response_status Int64,
+transfer_size Int64,
+""".strip()
+
+"""https://developer.mozilla.org/en-US/docs/Web/API/LargestContentfulPaint"""
+LARGEST_CONTENTFUL_PAINT_EVENT_COLUMNS = """
+largest_contentful_paint_element String,
+largest_contentful_paint_render_time Int64,
+largest_contentful_paint_load_time Int64,
+largest_contentful_paint_size Int64,
+largest_contentful_paint_id String,
+largest_contentful_paint_url String,
+""".strip()
+
+"""https://developer.mozilla.org/en-US/docs/Web/API/PerformanceNavigationTiming"""
+NAVIGATION_EVENT_COLUMNS = """
+dom_complete Int64,
+dom_content_loaded_event Int64,
+dom_interactive Int64,
+load_event_end Int64,
+load_event_start Int64,
+redirect_count Int64,
+navigation_type LowCardinality(String),
+unload_event_end Int64,
+unload_event_start Int64,
+""".strip()
+
+columns = ",".join(
+    [
+        BASE_PERFORMANCE_EVENT_COLUMNS.rstrip(","),
+        RESOURCE_EVENT_COLUMNS.rstrip(","),
+        LARGEST_CONTENTFUL_PAINT_EVENT_COLUMNS.rstrip(","),
+        NAVIGATION_EVENT_COLUMNS.rstrip(","),
+    ]
+)
+
+PERFORMANCE_EVENT_TABLE_ENGINE = lambda: ReplacingMergeTree(
+    "performance_events", ver="_timestamp", replication_scheme=ReplicationScheme.SHARDED
+)
+
+PERFORMANCE_EVENTS_TABLE_BASE_SQL = """
+CREATE TABLE IF NOT EXISTS {table_name} ON CLUSTER '{cluster}'
+(
+    {columns}
+    {extra_fields}
+) ENGINE = {engine}
+"""
+
+PERFORMANCE_EVENTS_TABLE_SQL = (
+    PERFORMANCE_EVENTS_TABLE_BASE_SQL
+    + """PARTITION BY toYYYYMM(origin_timestamp)
+ORDER BY (team_id, toDate(origin_timestamp), session_id, pageview_id)
```

Comment:
Problems: 
1. Looking up (only) via pageview_id with this `ORDER BY` is really really expensive.
2. Looking up via session_id requires inspecting more granules than needed.

Re 1, could we always include session_id in the where clause when looking up by pageview_id?

Re 2, could we also add a filter on origin_timestamp to all the point queries?